services/autorust/codegen/src/spec.rs

```diff
@@ -349,82 +350,129 @@ pub mod openapi {
-/// The module name is optional and is text before an underscore in the operatonId.
-fn function_name_from_operation_id(operation_id: &str) -> (Option<String>, String) {
-    let parts: Vec<&str> = operation_id.splitn(2, '_').collect();
-    if parts.len() == 2 {
-        (Some(parts[0].to_snake_case()), parts[1].to_snake_case())
-    } else {
-        (None, parts[0].to_snake_case())
+impl<'a> WebVerb {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            WebVerb::Get => "get",
+            WebVerb::Post => "post",
+            WebVerb::Put => "put",
+            WebVerb::Patch => "patch",
+            WebVerb::Delete => "delete",
+            WebVerb::Options => "options",
+            WebVerb::Head => "head",
+        }
     }
 }
 
 /// Creating a function name from the path and verb when an operationId is not specified.
 /// All azure-rest-api-specs operations should have an operationId.
-fn create_function_name(path: &str, verb_name: &str) -> String {
+fn create_function_name(verb: &WebVerb, path: &str) -> String {
     let mut path = path.split('/').filter(|&x| !x.is_empty()).collect::<Vec<_>>();
-    path.push(verb_name);
+    path.insert(0, verb.as_str());
     path.join("_")
 }
 
-pub fn path_item_operations(item: &PathItem) -> impl Iterator<Item = OperationVerb> {
+struct OperationVerb<'a> {
+    pub operation: Option<&'a Operation>,
+    pub verb: WebVerb,
+}
+
+pub fn path_operations(path: &str, item: &PathItem) -> Vec<WebOperation> {
     vec![
-        item.get.as_ref().map(OperationVerb::Get),
-        item.post.as_ref().map(OperationVerb::Post),
-        item.put.as_ref().map(OperationVerb::Put),
-        item.patch.as_ref().map(OperationVerb::Patch),
-        item.delete.as_ref().map(OperationVerb::Delete),
-        item.options.as_ref().map(OperationVerb::Options),
-        item.head.as_ref().map(OperationVerb::Head),
+        OperationVerb {
+            operation: item.get.as_ref(),
+            verb: WebVerb::Get,
+        },
+        OperationVerb {
+            operation: item.post.as_ref(),
+            verb: WebVerb::Post,
+        },
+        OperationVerb {
+            operation: item.put.as_ref(),
+            verb: WebVerb::Put,
+        },
+        OperationVerb {
+            operation: item.patch.as_ref(),
+            verb: WebVerb::Patch,
+        },
+        OperationVerb {
+            operation: item.delete.as_ref(),
+            verb: WebVerb::Delete,
+        },
+        OperationVerb {
+            operation: item.options.as_ref(),
+            verb: WebVerb::Options,
+        },
+        OperationVerb {
+            operation: item.head.as_ref(),
+            verb: WebVerb::Head,
+        },
     ]
     .into_iter()
-    .filter_map(|x| x)
+    .filter_map(|op_verb| match op_verb.operation {
+        Some(op) => {
+            let mut parameters = item.parameters.clone();
+            parameters.append(&mut op.parameters.clone());
+            Some(WebOperation {
```

Comment:
`WebOperation` in a step towards having a model to separate the OpenAPI v2 and Codegen.